### DISCUSSION TITLE
Configering Mocha isnt working

### DISCUSSION THREAD

**Discussion by HiranMadhu** [redhat-developer/vscode-extension-tester#1181] [Q&A] [answered]
I tried the mocha configuring technique as described and I want to get the output as in mocha. 
![image](https://github.com/redhat-developer/vscode-extension-tester/assets/83911435/0ea3a6b2-467d-4712-8da8-591256ca402a)

But the output is still the same as in extest. 
![image](https://github.com/redhat-developer/vscode-extension-tester/assets/83911435/5cab304b-963e-4e6d-8bf9-b0c338e4b20c)

can you help me on this. this is the mocha config file
`import { MochaOptions } from "vscode-extension-tester";

const options: MochaOptions = {
    reporter: 'spec',
    slow: 75,
    timeout: 2000,
    bail:false
}

export default options;`

**Comment by HiranMadhu** [comment]
This is the line says mocha loaded 
![image](https://github.com/redhat-developer/vscode-extension-tester/assets/83911435/a7260dbc-be84-4b2a-a0a7-0dd3c9b4dab7)

**Comment by djelinek** [comment, marked as answer]
Hello, I am not the exact Mocha expert but I have tried locally and for me the Mocha config file loaded by ExTester seems work as expected. 

by default Mocha is using `reporter: 'spec'` which output looks:
```
Launching tests...
  ProblemsView
    ✔ get all markers works (38ms)
    ✔ get warnings works (93ms)
    ✔ get errors works (75ms)
    ✔ get files works (89ms)
    ✔ filtering works (691ms)
    Marker
      ✔ getType works (78ms)
      ✔ getText works (75ms)
      ✔ toggleExpand works (265ms)
      ✔ click works (168ms)
```

when I used custom `.mochaarc.js` with content:
```js
module.exports = {
    reporter: 'list'
}
```

the output was changed to list as you can see below:
```
Launching tests...
  ✔ ProblemsView get all markers works: 54ms
  ✔ ProblemsView get warnings works: 99ms
  ✔ ProblemsView get errors works: 83ms
  ✔ ProblemsView get files works: 83ms
  ✔ ProblemsView filtering works: 702ms
  ✔ ProblemsView Marker getType works: 122ms
  ✔ ProblemsView Marker getText works: 102ms
  ✔ ProblemsView Marker toggleExpand works: 254ms
  ✔ ProblemsView Marker click works: 156ms
```

of course I do not know which reporter you need to use or implement to have the ouput looks like you shared in the picture but it seems to not be the bug of ExTester itself

**Comment by boonyashka** [comment]
@djelinek I have similar issue
Im trying to use `reporter: 'allure-mocha'` but it doesn't work (v5.5.2) -> for some reason extest continue use `reporter: 'spec'`
Can you check allure-mocha option?

**Reply by djelinek** [reply]
Hi, I've tested for `vscode-extension-tester-example` repository. Using [allure-mocha](https://www.npmjs.com/package/allure-mocha) NPM package and I am able to get a report HTML page just fine, see below:

<img width="927" alt="image" src="https://github.com/user-attachments/assets/a312c5dc-4f37-40a9-88ec-774ef7a4ea41" />



I am using `.mocharc.js` placed in a root of project which is loaded by Mocha runner automatically
```js
module.exports = {
	timeout: 5000,
	reporter: 'allure-mocha'
};
```
